### ISSUE TITLE
I should probably rewrite this

### ISSUE THREAD

**Issue by xxcodianxx** [xxcodianxx/youtube-dl-web#46]
This tool is outdated and sort of messily written (I'm looking at you DownloadPage.tsx). I hate React! The backend works, but it can be hacky at times (partially due to all the features people have contributed) and could be improved. I'm thinking of rewriting this sometime in the near future, but right now I don't have much time on my hands due to exams, so it will have to wait.

I still need to review PRs like #44 but right now I don't have time to think about the implications of supporting every platform that youtube-dl supports, and whether everything would work.

The new stack I'm thinking of for this app would probably be Svelte and either Python FastAPI or something in Rust (which could potentially end up being faster).

Thoughts?

**Comment by dazeb** [comment]
Good idea, changing to yt-dlp or adding it as a download option would be good similar to yt-dlp material.

**Comment by Elshara** [comment]
Look at supporting something like this?
https://github.com/Leonetienne/Tubio
https://github.com/kieraneglin/pinchflat
https://github.com/OrfiTeam/OrpheusDL
https://github.com/KraXen72/shira
https://github.com/nathom/streamrip
https://www.npmjs.com/package/lucida
It has all the features you want but the code base is completely different. These projects are the closest in scope that I can think of to this one if you're looking for ideas.
